src/qt/msgdlg.cpp

```diff
@@ -108,17 +117,14 @@ wxMessageDialog::wxMessageDialog( wxWindow *parent, const wxString& message,
 
     if ( style & wxSTAY_ON_TOP )
         dlg->setWindowModality( Qt::ApplicationModal );
-
-    PostCreation(false);
-
-    Centre(wxBOTH | wxCENTER_FRAME);
 }
 
-wxIMPLEMENT_CLASS(wxMessageDialog,wxDialog);
-
 int wxMessageDialog::ShowModal()
 {
     WX_HOOK_MODAL_DIALOG();
+
+    QtCreateMsgDialog();
```

Comment:
Why do we have to call this from here instead of creating it from the ctor as before? It looks like we could just call `setInformativeText()` here if necessary, couldn't we?

This results in (apparently) unnecessary changes and potentially creates problems, e.g. if `ShowModal()` is called twice.